modular_skyrat/modules/modular_implants/code/nifs.dm

```diff
@@ -0,0 +1,511 @@
+		TOOL_SCREWDRIVER = 15,
+	)
+	time = 12 SECONDS
+
+/datum/surgery_step/repair_nif/preop(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	display_results(user, target, span_notice("You begin to restore the integrity of [target]'s NIF..."),
+		"[user] begins to fix [target]'s NIF.",
+		"[user] begins to perform repairs on [target]'s NIF.")
+
+/datum/surgery_step/repair_nif/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	display_results(user, target, span_notice("You succeed in restoring the integrity of [target]'s NIF."),
+		"[user] successfully repairs [target]'s NIF!",
+		"[user] completes the repair on [target]'s NIF.")
+
+	var/mob/living/carbon/human/nif_patient = target
+	var/obj/item/organ/internal/cyberimp/brain/nif/installed_nif = nif_patient.installed_nif
+
+	installed_nif.durability = installed_nif.max_durability
+	installed_nif.send_message("Restored to full integrity!")
+
+	return ..()
+
+/datum/surgery_step/repair_nif/failure(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	if(target.getorganslot(ORGAN_SLOT_BRAIN))
+		display_results(user, target, span_warning("You screw up, causing [target] brain damage!"),
+			span_warning("[user] screws up, causing damage to [target]'s brain!"),
+			"[user] completes the repair on [target]'s NIF.")
+
+		target.adjustOrganLoss(ORGAN_SLOT_BRAIN, 20)
+	return FALSE
+
+// Action used to pull up the NIF menu
+/datum/action/item_action/nif
+	button_icon = 'modular_skyrat/master_files/icons/mob/actions/action_backgrounds.dmi'
+	background_icon_state = "android"
+	icon_icon = 'modular_skyrat/master_files/icons/mob/actions/actions_nif.dmi'
+	check_flags = AB_CHECK_CONSCIOUS
+
+/datum/action/item_action/nif/open_menu
+	name = "Open NIF Menu"
+	button_icon_state = "user" // This is a placeholder
+
+/datum/action/item_action/nif/open_menu/Trigger(trigger_flags)
+	. = ..()
+	var/obj/item/organ/internal/cyberimp/brain/nif/target_nif = target
+
+	if(target_nif.calibrating)
+		target_nif.send_message("The NIF is still calibrating, please wait!", TRUE)
+		return FALSE
+
+	if(target_nif.durability < 1)
+		target_nif.send_message("Durability low!", TRUE)
+		return FALSE
+
+	if(target_nif.broken)
+		target_nif.send_message("The NIF is unable to be used at this time!", TRUE)
+		return FALSE
+
+	if(!.)
+		return
+
+	target_nif.ui_interact(usr)
+
+/mob/living/carbon/human
+	///What text is shown upon examining a human with a NIF?
+	var/nif_examine_text
+	//What, if any NIF is currently installed inside of the mob?
+	var/obj/item/organ/internal/cyberimp/brain/nif/installed_nif
+
+/mob/living/carbon/human/examine(mob/user)
+	. = ..()
+
+	if(nif_examine_text)
+		. += nif_examine_text
+
+/mob/living/carbon/human/death(gibbed)
+	. = ..()
+
+	if(!installed_nif)
+		return
+
+	installed_nif.durability -= installed_nif.death_durability_loss
+	installed_nif.durability_loss_vulnerable = FALSE
```

Comment:
What happens when they get gibbed, though?

Reply:
I tested this, it shouldn't matter if they get gibbed or not.